### ISSUE TITLE
Shippable is frequently causing my Chrome tabs to crash

### ISSUE THREAD

**Issue by johnsonwj** [Shippable/support#3844]
### Description of your issue:

I have had Shippable tabs crash multiple times per day. It seems not to be connected to any particular actions:

* A tab containing an rSync log that was left alone for a while freezes and then has to be killed
* The SPOG view freezes and has to be killed
* The SPOG view freezes when trying to click on a button (flags, settings, views) and then has to be killed

This is severely affecting my ability to do work setting up a new repository with CI in Shippable.

Thank you!

**Comment by johnsonwj** [comment]
The SPOG has also been taking an extremely long time to respond to interactions like dragging the view or right-clicking on jobs, even when filtered down to only a couple flags and ~25 jobs visible.

**Comment by johnsonwj** [comment]
A coworker has also had this issue recently, so it is unlikely to be machine-specific.

**Comment by johnsonwj** [comment]
This is now happening extremely frequently, almost every time I open a tab.

**Comment by manishas** [comment]
@ambarish2012 will get in touch with you asap.

**Comment by johnsonwj** [comment]
He has, thank you!

**Comment by johnsonwj** [comment]
![image](https://user-images.githubusercontent.com/3478488/30612442-6b55ffac-9d39-11e7-9357-03b233fd78cd.png)

**Comment by avinci** [comment]
@johnsonwj, I apologize for the issues that you are facing. We did some analysis and we have determined that this occurs only on the SPOG view and when web sockets are actively firing when there is significant number of objects on the SPOG.

There are 3 reasons for this
1. D3.js is a super CPU intensive process and its causing the browser to become unresponsive under tremendous load
1. The load is significantly increasing due to sockets that the SPOG is rendering over and over again 
1. The reason for SPOG loading again and again is that we are reacting to web sockets that actually does not require the SPOG render again and again and this is core of the problem

For a short term workaround, switch to grid view and you should not see this problem when there are over 500+ objects on SPOG or you can use a custom view to reduce the number of objects and still use the SPOG view. 

In the meantime here is what we are doing to fix this permanently

1. We will push some fixes to reduce this problem in tomorrows deployment which might mitigate some of the unnecessary reloads of SPOG
1. Next week we are going to visit the D3.js logic and reduce the CPU intensive processes which should minimize this issue

 I hope the short term workaround is workable for you and I will get back to you once we have done the deployment tomorrow and we can test it out a bit and see if things have improved.

**Comment by johnsonwj** [comment]
@avinci thank you for the update! I'm sorry I did not see it until now.

I can try to use the grid view instead; however, I have tried using custom views limited to the pipeline I am working on (8-16 objects) and the tab will still become unresponsive and crash.

**Comment by johnsonwj** [comment]
@avinci 

Even in grid view, when switching back to the Shippable tab, the window is unresponsive and blank for several seconds.

**Comment by Jsgoller1** [comment]
FWIW this has been an issue since we started using Shippable too, and it happens to literally every browser/OS combination I've tried opening Shippable on. It even pegs out the CPU core running my browser on my high-end VR machine (i.e. Intel 6000-series i7 CPU, 32 GB of memory, etc). I can reliably expect the SPOG (and sometimes the CI tab) to hang my browser whenever I open Shippable.

**Comment by avinci** [comment]
Thx @Jsgoller1 

We have identified the problem and working on a fix. This will be deployed on Thursday and should resolve this issue.  TLDR: We are doing some unnecessary processing which needs to be removed

Long version, this is happening due to socket re-paint and unnecessary refresh of data. This is alleviated a bit with cache which we introduced to make it snappy, but it still is unnecessarily refreshing which causes angular to fire events which causes the browser to become unresponsive. 

I will let you once we have tested this fix and deployed and would love to benchmark this.
